fastplotlib/plot.py

```diff
@@ -103,4 +104,57 @@ def show(self, autoscale: bool = True):
         if autoscale:
             self.auto_scale(maintain_aspect=True, zoom=0.95)
 
-        return self.canvas
+        if toolbar:
+            tools = ToolBar(self).toolbar
```

Comment:
Make the toolbar an attribute of Plot so its only instantiated once. This would create new toolbars every time show is called which gets messy